lib/remote_persistent_term.ex

```diff
@@ -61,13 +61,18 @@ defmodule RemotePersistentTerm do
     ]
   ]
 
+  @type identifiers :: %{
+          etag: String.t(),
```

Comment:
why are we moving etag up to this module when it is currently only a concept in the s3 fetcher?

Reply:
well we were already returning it to this module, we just called it "version". Etag and version are different concepts and etag isn't just an s3 specific value, it could also be set for http cache servers too.

Reply:
we need the version id in order to specifically request a version in the retry logic of the code. We could move the etag check to inside of the s3 module (and not return the etag) but I thought that if we were to later on implement etags in other fetchers it would mean duplicating code, so I kept it here.